engine/src/main/java/org/terasology/world/block/SquareRegion.java

```diff
@@ -0,0 +1,671 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.world.block;
+
+import org.joml.AABBi;
+import org.joml.Math;
+import org.joml.Rectanglef;
+import org.joml.Rectanglei;
+import org.joml.RoundingMode;
+import org.joml.Vector2fc;
+import org.joml.Vector2i;
+import org.joml.Vector2ic;
+import org.joml.Vector3ic;
+
+/**
+ * is a bounded rectangle describing squares contained within.
+ * A {@link SquareRegion} is described and backed by an {@link org.joml.Rectanglei}
+ */
+public class SquareRegion {
```

Comment:
"Square" in the sense of 90° angles, or in the sense of a quadratic rectangle with all four sides of equal length?
I'd vote to call this `RectangularRegion` for clearer naming and to avoid confusion.